specs/lib/updateMaterialPropertiesSpec.js

```diff
@@ -0,0 +1,236 @@
+        };
+
+        var gltfWithBlendExtension = updateMaterialProperties(gltf);
+        expect(gltfWithBlendExtension.extensionsUsed.indexOf('EXT_blend')).toBeGreaterThan(-1);
+        expect(gltfWithBlendExtension.extensionsRequired.indexOf('EXT_blend')).toEqual(-1);
+
+        var material = gltfWithBlendExtension.materials[0];
+        expect(material.alphaMode).toBe('BLEND');
+        expect(material.extensions).toBeDefined();
+        var materialBlending = material.extensions.EXT_blend;
+        expect(materialBlending).toBeDefined();
+        expect(materialBlending.blendEquation).toEqual([
+            WebGLConstants.FUNC_ADD,
+            WebGLConstants.FUNC_ADD
+        ]);
+        expect(materialBlending.blendFactors).toEqual([
+            WebGLConstants.ONE,
+            WebGLConstants.ONE_MINUS_SRC_ALPHA,
+            WebGLConstants.ONE,
+            WebGLConstants.ONE_MINUS_SRC_ALPHA
+        ]);
+    });
+
+    it('does not set alphaMode or add EXT_blend if no blending is found in render states', function () {
+        var gltf = {
+            programs: {
+                program_0: {
+                    attributes: [
+                        'a_normal',
+                        'a_position',
+                        'a_texcoord0'
+                    ],
+                    fragmentShader: 'BoxTextured0FS',
+                    vertexShader: 'BoxTextured0VS'
+                }
+            },
+            shaders: {
+                BoxTextured0FS: {
+                    type: WebGLConstants.FRAGMENT_SHADER,
+                    uri: 'BoxTextured0FS.glsl'
+                },
+                BoxTextured0VS: {
+                    type: WebGLConstants.VERTEX_SHADER,
+                    uri: 'BoxTextured0VS.glsl'
+                }
+            },
+            techniques: {
+                technique0: {
+                    states: {
+                        enable: [
+                            WebGLConstants.DEPTH_TEST,
+                            WebGLConstants.CULL_FACE
+                        ],
+                        functions: {
+                            depthMask: false
+                        }
+                    }
+                }
+            },
+            materials: [
+                {
+                    technique: 'technique0',
+                    extensions : {
+                        KHR_techniques_webgl: {}
+                    }
+                }
+            ],
+            extensionsUsed: [],
+            extensionsRequired: []
+        };
+
+        var gltfWithBlendExtension = updateMaterialProperties(gltf);
+        expect(gltfWithBlendExtension.extensionsUsed.indexOf('EXT_blend')).toEqual(-1);
+        expect(gltfWithBlendExtension.extensionsRequired.indexOf('EXT_blend')).toEqual(-1);
+
+        var material = gltfWithBlendExtension.materials[0];
+        expect(material.alphaMode).toBeUndefined();
+        expect(material.extensions).toBeDefined();
```

Comment:
At first I was confused why the extensions were defined, and then noticed `KHR_techniques_webgl`. Maybe just remove that.